### PR TITLE
HV-1007 Using XMLReaderEvent#peek to look ahead for schema version in…

### PR DESCRIPTION
**Pull request by hferentschik**
…stead of mark&reset of stream

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/xml/XmlParserHelper.java [hunk 1]
```diff
@@ -72,21 +71,15 @@
 	 * represented by the "version" attribute of the root element of the stream.
 	 *
 	 * @param resourceName The name of the represented XML resource.
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/xml/XmlParserHelper.java]: The `READ_LIMIT` constant is not needed anymore.

**Comment by gunnarmorling** [comment]
Merged; I first was concerned about re-using the same reader for getting the version and subsequent unmarshalling, but it works fine as unmarshalling works fine from the start element retrieved for getting the version. Neat trick :)

Added one more commit for removing the unused constant. I couldn't test on IBM JDK but I assume you did so?

**Comment by hferentschik** [comment]
> Added one more commit for removing the unused constant.

nice

 > I couldn't test on IBM JDK but I assume you did so?

I did

File: engine/src/main/java/org/hibernate/validator/internal/xml/XmlParserHelper.java (hunk 1 above)
Comment (hferentschik) [inline engine/src/main/java/org/hibernate/validator/internal/xml/XmlParserHelper.java]: good catch

**Pull request merged**